src/lib.rs

```diff
@@ -171,6 +189,34 @@ pub trait Integer: Sized + Num + PartialOrd + Ord + Eq {
     fn div_mod_floor(&self, other: &Self) -> (Self, Self) {
         (self.div_floor(other), self.mod_floor(other))
     }
+
+    /// Rounds up to nearest multiple of argument.
+    ///
+    /// # Examples
+    ///
+    /// ~~~
+    /// # use num_integer::Integer;
+    /// assert_eq!(16.round_up_to(&8), 16);
+    /// assert_eq!(23.round_up_to(&8), 24);
+    /// ~~~
+    #[inline]
+    fn round_up_to(&self, other: &Self) -> Self where Self: traits::NumRef {
```

Comment:
> I think it could use "multiple" in the name -- maybe `next_multiple`?

I now called them `next_multiple_of` and `next_multiple_back_of` (for consistency with `Iterator` vs `DoubleEndedIterator`).

> How should negative self and/or other behave?

I'd think it ought to go the other way:
`self.next_multiple_of(other) = self.next_multiple_back_of(other.neg())`

The new definition fulfills this criterion. I added some tests too.

> I think I'd rather just `clone()` instead of introducing NumRef here

Done

Reply:
I don't care for `next_multiple_back_of` -- "back" makes more sense for iterators than for numbers. I guess you can thing of it like range iterators, but it still feels clumsy. How about `previous_multiple_of` or `prev_multiple_of`?

Please test with negative `self` values. We should explain in words too, that rounding up/down is reversed with negative `other`.

Reply:
I think `previous_multiple_of` is too long, and the abbreviation `prev` could be ambiguous — `preview` is the clash which readily comes to mind, but i'm not sure how relevant it is here. How about `past_multiple_of`?

Reply:
I like `prev` better than `past`. I don't see any precedent in the `std` API supporting that, but there's a lot of use of `prev_` meaning previous throughout rust's implementation. So I think that's well understood, especially in context with `next_`.